dbt/adapters/redshift/connections.py

```diff
@@ -323,6 +327,68 @@ def __iam_idc_browser_kwargs(credentials) -> Dict[str, Any]:
 
         return __iam_kwargs(credentials) | idc_kwargs
 
+    def __iam_idc_token_kwargs(credentials) -> Dict[str, Any]:
+        """
+        Expand the logic and flow control here to account for other Identity providers
+        with their specific request configurations. May require new parameters for
+        token_endpoint extraction to distinguish idp's if the url is insufficient.
+
+        We only support token_type=EXT_JWT tokens. token_type=ACCESS_TOKEN has not been
+        tested or is supported but can be added with a presenting use-case.
+        """
+
+        def handle_response(response):
+            """
+            Rate limiting of access_token generation has been called out during refinement
+            and could require special handling, so this method exists to encapsulate that
+            code with an eye towards future handling. All other failures are thrown as
+            normal request failures.
+            """
+            # Handle the 429 rate-limiting case first
+            if response.status_code == 429:
```

Comment:
if we're hitting the rate limit could we add an exponential backoff/retry strategy here?

Reply:
haha, I kind of knew you were going to prompt me for that. We're actively determining that customer experience. Let me filter this up.

Reply:
What are the scenarios where we're hitting the rate limit? I don't see a retry on this, so is this using some generalized retry? Does this get run on each model? Each run?

Reply:
If we actually do a retry strategy -- which it's unclear and trending towards no based on @jenniferjsmmiller 's current research -- it'd be every connection. The 429 is specifically a rate limiting error status from okta

Reply:
Since we're reusing connections where possible, this feels like a low occurrence event. In other words, if we hit the rate limit, my guess is that something else is actually going wrong or the user is simply abusing the application (e.g. running a lot of threads of dbt-redshift in parallel via something like airflow, which isn't officially supported).

Reply:
We don't reuse redshift connections do we? And also there was concern about 100 threads being set and this occuring at startup but even that was something Jenn didn't seem to trigger in their research 🤞